src/PersonObjects/Player/PlayerObjectAugmentationMethods.ts

```diff
@@ -6,11 +6,29 @@ import { updateGoMults } from "../../Go/effects/effect";
 import type { PlayerObject } from "./PlayerObject";
 
 export function applyEntropy(this: PlayerObject, stacks = 1): void {
+  // Save the current value of this.hp.
+  const currentHp = this.hp.current;
+  const currentMaxHp = this.hp.max;
+
   // Re-apply all multipliers
   this.reapplyAllAugmentations();
   this.reapplyAllSourceFiles();
 
   this.mults = calculateEntropy(stacks);
   staneksGift.updateMults();
   updateGoMults();
+
+  if (this.hp.max === currentMaxHp) {
```

Comment:
Simplifications: I don't think you need this `if` at all. If the max is unchanged, the round below will ensure that the same hp value is calculated. 

Given that, you can calculate and save the ratio at the top, which makes both the expression and explanation simpler.